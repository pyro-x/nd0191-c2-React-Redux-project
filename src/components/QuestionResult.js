const QuestionResult = ({ optionText, winner, votes, percentage }) => {

    const winnerCLass = winner ? "bg-green-500" : " bg-gray-500";
    return (
            <div className='flex flex-col space-y-4 p-4 border shadow w-1/2 justify-between'>
                <div>{optionText}</div>
                <div> Number of Votes {votes} </div>
                <div className={winnerCLass + " text-white font-bold py-2 px-4 rounded mt-8 disabled:opacity-50 text-center"}>{percentage}%</div>
            </div>
    )
}

export default QuestionResult;