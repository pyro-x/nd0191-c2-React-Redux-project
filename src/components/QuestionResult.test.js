import { render } from '@testing-library/react'

import QuestionResult  from './QuestionResult'
import * as React from 'react';

describe('QuestionResult', () => {
    it ('renders correctly', () => {
        const  view  = render(
            <QuestionResult optionText="snapshot test" winner={true} votes={1} percentage={100}/>
            );
        expect(view).toMatchSnapshot()
    });
});