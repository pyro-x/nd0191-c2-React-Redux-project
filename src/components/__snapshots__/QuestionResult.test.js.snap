// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuestionResult renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex flex-col space-y-4 p-4 border shadow w-1/2 justify-between"
      >
        <div>
          snapshot test
        </div>
        <div>
           Number of Votes 
          1
           
        </div>
        <div
          class="bg-green-500  text-white font-bold py-2 px-4 rounded mt-8 disabled:opacity-50 text-center"
        >
          100
          %
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex flex-col space-y-4 p-4 border shadow w-1/2 justify-between"
    >
      <div>
        snapshot test
      </div>
      <div>
         Number of Votes 
        1
         
      </div>
      <div
        class="bg-green-500  text-white font-bold py-2 px-4 rounded mt-8 disabled:opacity-50 text-center"
      >
        100
        %
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
